streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1148,6 +1148,9 @@ private void verifyMaxInFlightRequestPerConnection(final Object maxInFlightReque
         consumerProps.put(REPLICATION_FACTOR_CONFIG, getInt(REPLICATION_FACTOR_CONFIG));
         consumerProps.put(APPLICATION_SERVER_CONFIG, getString(APPLICATION_SERVER_CONFIG));
         consumerProps.put(NUM_STANDBY_REPLICAS_CONFIG, getInt(NUM_STANDBY_REPLICAS_CONFIG));
+        consumerProps.put(ACCEPTABLE_RECOVERY_LAG_CONFIG, getLong(ACCEPTABLE_RECOVERY_LAG_CONFIG));
+        consumerProps.put(MAX_WARMUP_REPLICAS_CONFIG, getInt(MAX_WARMUP_REPLICAS_CONFIG));
+        consumerProps.put(PROBING_REBALANCE_INTERVAL_MS_CONFIG, getLong(PROBING_REBALANCE_INTERVAL_MS_CONFIG));
```

Comment:
Can we add a note to `AssignorConfiguration` explaining that you have to do this any time you add a new Streams config?

Reply:
Yes, we copy over the other internal configs in org.apache.kafka.streams.processor.internals.StreamThread#create

I'll add your listener to the config test. I'm not sure about PartitionGrouper.

Reply:
The answer about your listener is that we automatically pass through any config that _isn't_ registered. The assumption is, I suppose, that users might want to pass through custom configs to their custom plugged-in components.

The configs in this PR got filtered out because they are registered configs in the public API.

Reply:
Turns out PartitionGrouper does _not_ get copied over. I'll create a jira to track this, so that we don't have to get sidetracked in this PR with the question of what we should do about this deprecated config.

Reply:
https://issues.apache.org/jira/browse/KAFKA-10046

Reply:
> we automatically pass through any config that isn't registered

I have to say, this seems totally backwards to me. So basically we just happen to pass in any number of configs that we may or may not need, but will split out specific configs that we _do_ need unless explicitly told to include them? I understand the custom configs motivation, but then why not just pass through all the configs?

What if I wanted to access the value of one of my registered Streams configs in my plugged-in component? I'd have to add the same value a second time, but with an unregistered config name. Huh?

Reply:
Regarding KAFKA-10046, in current trunk we already have some logic that assumes the default partition grouper is always used, so I'd suggest we just bite the bullet and remove it in 2.6.